app/src/main/java/com/genymobile/gnirehtet/GnirehtetControlReceiver.java

```diff
@@ -63,6 +63,13 @@ public void onReceive(Context context, Intent intent) {
             startGnirehtet(context, config);
         } else if (ACTION_GNIREHTET_STOP.equals(action)) {
             stopGnirehtet(context);
+        } else if ("android.hardware.usb.action.USB_STATE".equals(action)){
```

Comment:
IMO, it should rely on whether the socket is connected or not, whatever the USB state is (the socket can be disconnected while the USB cable is still connected).

Reply:
Oh, i not replace intent "ACTION_GNIREHTET_STOP"("socket can be disconnected while the USB cable is still connected"), i add intent USB_STATE with disconnected (if cabel unplug, then socket dont work too).

Reply:
It should not rely on USB state, but on the socket connection state.